rx-preferences/src/main/java/rx/android/preferences/Preference.java

```diff
@@ -5,58 +5,91 @@
 import rx.functions.Action1;
 import rx.functions.Func1;
 
-import static rx.android.preferences.Utils.isNullOrEmpty;
+/** A preference of type {@link T}. Instances can be created from {@link RxSharedPreferences}. */
+public final class Preference<T> {
+  /** Stores and retrieves instances of {@code T} in {@link SharedPreferences}. */
+  public interface Adapter<T> {
+    /**
+     * Retrieve the value for {@code key} from {@code preferences}, using {@code defaultValue} if
+     * no value exists.
+     */
+    T get(String key, T defaultValue, SharedPreferences preferences);
 
-/**
- * Wraps a preference of type {@link T}.
- * </p>
- * Clients don't need to use this class directly, they'll use one of it's implementations instead.
- */
-public abstract class Preference<T> {
-  final SharedPreferences sharedPreferences;
-  final String key;
-  final T defaultValue;
+    /** Store nullable {@code value} for {@code key} in {@code editor}. */
+    void set(String key, T value, SharedPreferences.Editor editor);
```

Comment:
We should document that is not the Adapter's responsibility to call `commit` or `apply` on the `SharedPreferences.Editor`.